File: admin/src/models/order.js
export class OrderRequest {
  constructor() {
    this.shippingCost = 0;
    this.amount = 0;
    this.status = '';
    this.shippingAddress = '';
    this.userId = 0;
    this.orderItems = [];
  }
}

export class OrderUpdateRequest {
  constructor() {
    this.shippingCost = 0;
    this.status = '';
    this.shippingAddress = '';
  }
}
/////
export class OrderResponse {
  constructor() {
    this.id = 0;
    this.orderNumber = '';
    this.shippingCost = 0;
    this.amount = 0;
    this.status = '';
    this.createdAt = null;
    this.shippingAddress = '';
    this.user = new UserOrderResponse();
    this.orderItems = [];
  }
}
////
export class UserOrderResponse {
  constructor() {
    this.id = 0;
    this.name = '';
    this.phone = '';
  }
}

File: api/ZooOnlineStoreApi/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ZooOnlineStoreApi.Models
{
    public class Order
    {
        public int Id { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public decimal ShippingCost { get; set; } = 0;
        public decimal Amount { get; set; } = 0;
        public string Status { get; set; } = string.Empty;//pending, paid, processing, shipped, delivered, deleted
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public string ShippingAddress { get; set; } = string.Empty;
        //связи 
        public Payment? Payment { get; set; }//один к одному связь
        public int UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public User? User { get; set; }
        public HashSet<OrderItem>? OrderItems { get; set; }
        public Order() { }
    }
}
